aws/resource_aws_ram_resource_association.go

```diff
@@ -0,0 +1,210 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ram"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsRamResourceAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRamResourceAssociationCreate,
+		Read:   resourceAwsRamResourceAssociationRead,
+		Delete: resourceAwsRamResourceAssociationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_share_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRamResourceAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+	resourceARN := d.Get("resource_arn").(string)
+	resourceShareARN := d.Get("resource_share_arn").(string)
+
+	input := &ram.AssociateResourceShareInput{
+		ClientToken:      aws.String(resource.UniqueId()),
```

Comment:
I'm not sure what this has to do with an AWS account ID and we don't tend to re-document the AWS Go SDK structs inline. Using [`resource.UniqueId()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#UniqueId) is just for a 26 digit randomly generated number behind a mutex to help protect from concurrency issues. We do this rather than implementing our function when one is readily available. We could use [`resource.PrefixedUniqueId()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#PrefixedUniqueId) which would allow for some form of prefix beyond the `resource.UniqueId()` default of `"terraform-"`, etc. but many of these `ClientToken`/`IdempotencyToken` fields have byte length restrictions so in my opinion seems safer to just default to a short standard.

If you would like anything changed in this regard, we should create an issue and apply the change across the codebase.

Reply:
> If you would like anything changed in this regard, we should create an issue and apply the change across the codebase.

Sorry for the confusion I was actually referring to the field name `ClientToken`. At first glance it seemed like an AWS specific thing but after reading the documentation I understood why `resource.UniqueId()` was being used. No action necessary.